src/test/regress/sql/with.sql

```diff
@@ -1228,6 +1228,18 @@ JOIN cte b USING (i);
 
 DROP TABLE with_test;
 
+-- Test shared modifying CTE behavior in utility mode
+CREATE TABLE with_test (i int, j int) DISTRIBUTED RANDOMLY;
+
+-- check when slice is not allocated (topslice is allocated only on QD)
+\! PGOPTIONS='-c gp_session_role=utility' psql -d regression -c 'EXPLAIN (SLICETABLE, COSTS OFF) WITH cte AS (INSERT INTO with_test VALUES (1, 2) RETURNING *) SELECT i FROM cte a JOIN cte b USING (i);'
+\! PGOPTIONS='-c gp_session_role=utility' psql -d regression -c 'WITH cte AS (INSERT INTO with_test VALUES (1, 2) RETURNING *) SELECT i FROM cte a JOIN cte b USING (i);'
+
+-- check when slice is allocated inside InitPlan
+\! PGOPTIONS='-c gp_session_role=utility' psql -d regression -c 'EXPLAIN (SLICETABLE, COSTS OFF) WITH cte AS (INSERT INTO with_test VALUES (3, 4) RETURNING *) SELECT i FROM cte WHERE 4 = (SELECT j FROM cte);'
+\! PGOPTIONS='-c gp_session_role=utility' psql -d regression -c 'WITH cte AS (INSERT INTO with_test VALUES (3, 4) RETURNING *) SELECT i FROM cte WHERE 4 = (SELECT j FROM cte);'
```

Comment:
I think that queries should be executed not on the coordinator, see how it is done at the end of the test `src/test/regress/sql/bfv_dml_rpt.sql`